core/services/functions/listener.go

```diff
@@ -484,14 +604,21 @@ func (l *FunctionsListener) handleRequest(ctx context.Context, requestID Request
 	requestIDStr := formatRequestId(requestID)
 	l.logger.Infow("processing request", "requestID", requestIDStr)
 
+	err := l.validateRequestSignature(requestID, subscriptionOwner, requestData)
```

Comment:
Couldn't you just have "if v1 then validate" here? Why duplicating the whole method?